### PR TITLE
feat: add `--custom-resolver-options` to `bundle` command

### PR DESCRIPTION
**Pull request by szymonrybczak**
## Summary:

Added `--custom-resolver-options` to `--bundle` command. This options is also [available](https://github.com/facebook/metro/blob/main/docs/CLI.md#options) in Metro's CLI.

## Changelog:

[INTERNAL] [ADDED] - Add `--resolver-options` to `bundle` command

## Test Plan:

1. Build all packages by running `yarn build` in the root
2. Go to `packages/rn-tester` and run `npx react-native bundle --custom-resolver-options key=value` and the options should be passed to the Config.

### CODE REVIEW

**Comment by szymonrybczak** [comment]
cc @huntie

File: packages/community-cli-plugin/src/commands/bundle/buildBundle.js [hunk 1]
```diff
@@ -99,6 +100,7 @@ async function buildBundleWithConfig(
     minify: args.minify !== undefined ? args.minify : !args.dev,
     platform: args.platform,
     unstable_transformProfile: args.unstableTransformProfile,
+    customResolverOptions: args.resolverOptions,
```
Comment (thymikee) [inline packages/community-cli-plugin/src/commands/bundle/buildBundle.js:103]: I'd name it the same as the Metro's flag to avoid ambiguity

Comment (szymonrybczak) [reply packages/community-cli-plugin/src/commands/bundle/buildBundle.js]: 👍

File: packages/community-cli-plugin/src/commands/bundle/index.js [hunk 2]
```diff
@@ -114,6 +114,17 @@ const bundleCommand: Command = {
...
     },
+    {
+      name: '--custom-resolver-options <string>',
+      description: 'Custom resolver options, format: key=value,key2=value2.',
+      parse: (val: string): Record<string, string> => {
+        return val.split(',').reduce((options, option) => {
+          const [key, value] = option.split('=');
+          options[key] = value;
+          return options;
+        }, {});
```
Comment (huntie) [inline packages/community-cli-plugin/src/commands/bundle/index.js:125]: Flow needs the type of the initial value in this `reduce` call, i.e. a typecast `({}: Record<string, string>)`.

Can I suggest this alternative? (**Please test in practice**)

```suggestion
        return Object.fromEntries(
          val.split(',').map(option => {
            const [key, value] = option.split('=');
            return [key, value];
          }),
        );
```

**Pull request closed without merging**